File: 13주차/KM/83-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode cur = head;
        while(cur != null){
            ListNode temp = cur.next;
            if(temp == null) {
                cur.next = temp;
                break;
            }
            if(cur.val == temp.val){
                while(temp != null){
                    if(cur.val != temp.val){
                        break;
                    }
                    temp = temp.next;
                }
                cur.next = temp;
            }
            cur = cur.next;
        }
        return head;
    }
}

File: 13주차/DE/83. Remove Duplicates from Sorted List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 function deleteDuplicates(head) {
    pre = head
    if(head == null){
        return head
    }
    node = head.next
    
    while(node != null){
        if(pre.val == node.val){
            pre.next = node.next
        }
        else{
            pre = node
        }
        node = node.next
    }

    return head;
};